### DISCUSSION TITLE
Emulator experience

### DISCUSSION THREAD

**Discussion by Mellvik** [jbruchon/elks#1439] [General]
It seems reasonable to move the emulator discussion to its own thread. Hopefully there will be a lot more experience to share and discuss in the next weeks or months.

I've found that the Compaq Portable emulation (documented in the wiki) is not a good choice when used with the internal display controller. The cursor is always gone after bootup, and there seem to be no way to get it back. 
So while this display choice is the most stable and pleasant to use, the lack of a text cursor makes it useless. Other CGA alternatives work, but have other isses such as parts of the console window blinking. IOW the 2nd alternative works better with the listed config file.

-M

**Comment by ghaerr** [comment]
Hello @Mellvik,

Do you think we should submit a bug report to the 86box maintainers about the Compaq cursor, or perhaps look into fixing it ourselves? Given the cursor works for other configurations, this might not be that big a deal to implement.

**Comment by Mellvik** [comment]
Hi @ghaerr,

I've been considering that. The list of open issues on the 86Box project counts more than 200 and the response I got when queuing about the speed issue was not encouraging. OTOH we don't get anywhere if we don't report, so I decided to wait till I have a small list of issues instead of posting each one.

As to fixing it ourselves, that may be an educating endeavour indeed. I did fork and clone the project and had a brief look - although for a different reason: I tried to figure out how to get serial going. I gave up and asked on their Discord channel instead - which clarified the issue. NoGo.

Anyway, it may still be an easy fix, as you suggest. I suspect it has to do with timing: The cursor is there at boot time, during memory check, which seems to be disconnected from the specified (in the config file) clock frequency. Changing the clock from 6 to 25 MHz does not change the speed of the memory test, which it should. I haven't bothered yet to take the clock down to 6 to see if the cursor comes back. It's an easy test, but the system is unusable at that speed and I've been looking for useable more that bug hunting these past days.

IOW - 
- I do think it's worthwhile to submit an issue on github
- More testing is warranted to be as precise as possible in the problem report
- It's tempting to dive into the code to at least look around and get a feel for what this beast is about.

That said, I've admittedly spent more time with PCE these last few days. It's intriguingly simple, the serial connection solution is very good (actually, using a raw tcp connection we should be able to cat files back and forth between the host and the emulated system, and the ppp connection may be a very attractive setup for an ELKS ppp implementation which is on my schedule. Even the slowness (as in true emulation) of the system may be good in that context.

-M
